freezegun/api.py

```diff
@@ -230,35 +230,21 @@ def __call__(self, func):
         return self.decorate_callable(func)
 
     def decorate_class(self, klass):
-        klass_mro = klass.mro()
-        for attr in dir(klass):
-            if attr.startswith("_"):
-                continue
+        seen = set()
+        for base_klass in klass.mro():
+            for (attr, attr_value) in base_klass.__dict__.items():
+                if attr.startswith('_') or attr in seen:
+                    continue
+                seen.add(attr)
 
-            attr_value = getattr(klass, attr)
-            if not callable(attr_value):
-                continue
+                if not callable(attr_value) or inspect.isclass(attr_value):
+                    continue
 
-            for base_klass in klass_mro:
                 try:
-                    attr_descriptor = base_klass.__dict__[attr]
-                except KeyError:
-                    pass
-                else:
-                    break
-            else:
-                # don't sweat the attribute magic
-                attr_descriptor = None
-
-            # Skip patching classmethods and staticmethods
-            if isinstance(attr_descriptor, (classmethod, staticmethod)):
```

Comment:
I'm a bit confused why we are removing this part? Am I missing that this would now be redundant due to something else?